### PR TITLE
Groundwork for Streaming Progress Updates from View Service

### PR DESCRIPTION
**Pull request by hdevalence**
Sketch of one possibility for #3549 for discussion. (No code changes as yet, only proto changes).

### CODE REVIEW

File: proto/penumbra/penumbra/view/v1alpha1/view.proto
Comment (grod220) [inline proto/penumbra/penumbra/view/v1alpha1/view.proto:150]: Looks like these can be unnested so it can be shared with WitnessAndBuildResponse too

File: proto/penumbra/penumbra/view/v1alpha1/view.proto
Comment (grod220) [inline proto/penumbra/penumbra/view/v1alpha1/view.proto:140]: Does this correlate to the % of actions that have been built thus far?

**Review by grod220 — DISMISSED** [review]
Just two comments from my end

**Comment by hdevalence** [comment]
There's something off with the CI, this PR is green but it should definitely be red, it's half-finished and only changes protos, doesn't recompile them or propagate changes through the rest of the codebase. @conorsch any ideas?

File: proto/penumbra/penumbra/view/v1alpha1/view.proto
Comment (hdevalence) [reply proto/penumbra/penumbra/view/v1alpha1/view.proto:140]: It's up to the implementation, I guess, in principle an implementation that had insight into how much proving was done as it was happening could give better updates. I believe there's a way to get Rayon to do this on the Rust side. I think at the proto level we should be as agnostic as possible, and initially, we should just try to expose info we have rather than adding progress plumbing into the proving internals.

File: proto/penumbra/penumbra/view/v1alpha1/view.proto
Comment (hdevalence) [reply proto/penumbra/penumbra/view/v1alpha1/view.proto:150]: Sounds good, feel free to pick it up, we'll also have to propagate these changes through the rest of the source code.

**Comment by conorsch** [comment]
Good catch, I think I broke this back in https://github.com/penumbra-zone/penumbra/pull/3297, which altered the behavior so that the buf checks were run only when the PR is first opened, but not every time it's changed. Updated the logic in #3575, once that's in let's rebase this one and confirm we see the buf checks again!

**Comment by aubrika** [comment]
Blocked on #3621

**Comment by conorsch** [comment]
CI is now passing on this PR, module the protobuf lint failure, which is to be expected, because the proto changes here are Breaking Changes™. Will continue investigation of sweep test in #3621. Proceeding with merge here to unblock web impl consuming this new interface.

**Pull request merged**
